compair/static/modules/answer/answer-module.js

```diff
@@ -159,7 +159,10 @@ module.controller(
                     $scope.classlist = ret.objects;
                 }
             );
-            $scope.answer.user_id = $scope.loggedInUserId;
+            // only preset user_id when creating new answer
+            if ($scope.method == "create") {
+                $scope.answer.user_id = $scope.loggedInUserId;
```

Comment:
Can you instead move it below the object assignment in the other `if ($scope.method == "create")` on line 129. I think it will be easier to keep track of up there

ie:
```
if ($scope.method == "create") {
    $scope.answer = {
        ...
    };
    if ($scope.canManageAssignment) {
        $scope.answer.user_id = $scope.loggedInUserId;
    }
    ...
}
```